src/universal/components/Button/Button.js

```diff
@@ -149,18 +150,13 @@ class Button extends Component {
     const makeIconLabel = () => {
       const defaultIconPlacement = icon && label ? 'left' : '';
       const thisIconPlacement = iconPlacement || defaultIconPlacement;
-      const iconPlacementStyle = css(
+      const iconStyle = css(
+        styles.icon,
         thisIconPlacement === 'left' && styles.iconLeft,
         thisIconPlacement === 'right' && styles.iconRight,
       );
-      const iconMargin = iconOnly ? '' : iconPlacementStyle;
-      const iconStyle = {
-        fontSize: ui.buttonIconSize[size],
-        lineHeight: 'inherit',
-        verticalAlign: 'middle'
-      };
       const makeIcon = () =>
-        <FontAwesome className={iconMargin} name={icon} style={iconStyle} />;
+        <FontAwesome className={iconStyle} name={icon} />;
```

Comment:
I like how you cleaned up the styles here. Sometimes I’ve used inline styles for a few props when I’ve had conflicts with the Font Awesome classes. In this case the specificity works out just fine.

Reply:
Yeah, I was wondering about why this was implemented in this way. Figured there could be a good reason, so I was cautious with this change, but fortunately it seems to work!

Anyways, this inline styling / CSS-in-JS stuff is new to me, so I'm all ears as to discovering best practices.

Reply:
oh man i'm late to the party here. This works fine in development, but usually fails in production because in production the font-awesome stylesheet can take awhile to load & if it loads after the aphrodite stylesheet gets written then it'll take over. So, our options are (in order of ease of implementation): inline styles, `!important` all the things for that class, make our own copy of the font-awesome.css so we control load order, or move from aphrodite to something that doesn't load styles async.

Reply:
Just trying to understand a bit better here - could you clarfy what you mean by the font-awesome stylesheet will take over? From what I understand about the `Fontawesome` component and `aphrodite`, the rendered element gets both the font awesome class names (e.g. `fa fa-copy`) and the aphrodite class name (e.g. `some_weird_string011_abcdef`). So shouldn't it be the case that regardless of load order, the appropriate styles get applied?

Reply:
correct! it gets both. so let's say font-awesome says `font-size: 16px` and aphrodite says 20px. the css protocol is to pick the one with the selector that is most specific, and in that case, we have a tie where both are equally specific. So in that case, it's a race condition. aphrodite's solution is to suffix everything with `!important`, but that felt a little messier. here's a better description: https://github.com/Khan/aphrodite/pull/102

Reply:
Ah, got it, we're talking about conflicting properties! Ok cool. I'll create another PR correct this.

Reply:
Ah thanks for the good catch and the clarification @mattkrick yeah—hence when I work with FA I typically have had an inline style object for the properties that FA declares so that the specificity is greater regardless on when the CSS file loads. I was looking at it locally so everything loaded a-okay.